src/Core/Feature.js

```diff
@@ -0,0 +1,260 @@
+     * The sub geometry stored in indices, see constructor for more information.
+     * @param {number} count count of vertices
+     */
+    startSubGeometry(count) {
+        const last = this.indices.length - 1;
+        const offset = last > -1 ?
+            this.indices[last].offset + this.indices[last].count :
+            this._feature.vertices.length / this.size;
+        this.indices.push({ offset, count });
+        _extendBuffer(this._feature, count);
+    }
+
+    /**
+     * After you have pushed new the coordinates of sub geometry without <code>startSubGeometry</code>,
+     * this function close sub geometry. The sub geometry stored in indices,
+     * see constructor for more information.
+     * @param {number} count count of vertices
+     */
+    closeSubGeometry(count) {
+        const last = this.indices.length - 1;
+        const offset = last > -1 ?
+            this.indices[last].offset + this.indices[last].count :
+            this._feature.vertices.length / this.size - count;
+        this.indices.push({ offset, count });
+    }
+    /**
+     * Push new coordinates in vertices buffer.
+     * @param {Coordinates} coord The coordinates to push.
+     */
+    pushCoordinates(coord) {
+        if (coord.crs !== this._feature.crs) {
+            coord.as(this._feature.crs, coord);
+        }
+        if (this._feature.normals) {
+            coord.geodesicNormal.toArray(this._feature.normals, this._feature._pos);
+        }
+
+        this._feature._pushValues(this._feature, coord._values[0], coord._values[1], coord._values[2]);
+        // expand extent if present
+        if (this.extent) {
+            this.extent.expandByCoordinates(coord);
+        }
+    }
+
+    /**
+    * Push new values coordinates in vertices buffer.
+    * No geographical conversion is made or the normal doesn't stored.
+    *
+    * @param {number} value0 The longitude coordinate.
+    * @param {number} value1 The latitude coordinate.
+    * @param {number} value2 The altitude coordinate.
+    */
+    pushCoordinatesValues(value0, value1, value2) {
+        this._feature._pushValues(this._feature, value0, value1, value2);
+        // expand extent if present
+        if (this.extent) {
+            this.extent.expandByValuesCoordinates(value0, value1, value2);
+        }
+    }
+    /**
+     * @returns {number} the number of values of the array that should be associated with a coordinates.
+     * The size is 3 with altitude and 2 without altitude.
+     */
+    get size() {
+        return this._feature.size;
+    }
+}
+
+function push2DValues(feature, value0, value1) {
+    feature.vertices[feature._pos++] = value0;
+    feature.vertices[feature._pos++] = value1;
+}
+
+function push3DValues(feature, value0, value1, value2) {
+    feature.vertices[feature._pos++] = value0;
+    feature.vertices[feature._pos++] = value1;
+    feature.vertices[feature._pos++] = value2;
+}
+
+const FEATURE_TYPES = ['point', 'line', 'polygon'];
```

Comment:
I was thinking something more like

```js
export const FEATURE_TYPES = {
    POINT: 'point',
    MULTIPOINT: 'multipoint',
    LINESTRING: 'linestring',
    ....
}   
```

But if you are happy with this, it is ok for me.

Reply:
done